apps/pay/components/ParsePOSPayment/Receive-Invoice.tsx

```diff
@@ -32,9 +32,11 @@ const USD_MAX_INVOICE_TIME = 5 // minutes
 const PROGRESS_BAR_MAX_WIDTH = 100 // percent
 
 function ReceiveInvoice({ recipientWalletCurrency, walletId, state, dispatch }: Props) {
-  const deviceDetails = window.navigator.userAgent
-  const router = useRouter()
-  const { username, amount, unit, sats, memo } = router.query
+  const deviceDetails = window.navigator?.userAgent
+  const searchParams = useSearchParams()
+  const { username } = useParams()
+  const query = searchParams ? Object.fromEntries(searchParams.entries()) : {}
```

Comment:
I don't think we're using useSearchParams the way it should be

https://nextjs.org/docs/app/api-reference/functions/use-search-params

we should do more something like 

```
const amount = searchParams.get("amount")
const unit = searchParams.get("unit")
...
```

Reply:
We are creating an object from search parameters here. I think it is much cleaner and shorter due to destructuring ?
https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/entries